### PR TITLE
Add ENABLE_WHEEL=ON by default

### PR DESCRIPTION
**Pull request by slyubimt**
### Details:
 - *set ENABLE_WHEEL=ON by default*
 - *Add check for `wheel` Python package*

### Tickets:
 - *72568*

### CODE REVIEW

File: src/bindings/python/wheel/CMakeLists.txt [hunk 1]
```diff
@@ -18,10 +18,12 @@ if(LINUX)
                       NAMES patchelf
                       DOC "Path to patchelf tool")
     if(NOT patchelf_program)
-        message(FATAL_ERROR "patchelf is not found, which is needed to build ie_wheel")
+        message(FATAL_ERROR "patchelf is not found. It is required to build OpenVINO Runtime wheel")
     endif()
 endif()
 
+ie_check_pip_package(wheel FATAL_ERROR)
```
Comment (ilya-lavrenov) [inline src/bindings/python/wheel/CMakeLists.txt:25]: 1. should we also check `setuptools>=53.0.0`?
2. why don't we check exact version of `wheel`. I see we have `wheel>=0.36.2` in `requirements.txt`
3. should we check the whole file like `ie_check_pip_package(${CMAKE_CURRENT_SOURCE_DIR}/requirements-dev.txt FATAL_ERROR)`

File: cmake/features.cmake [hunk 2]
```diff
@@ -45,15 +45,15 @@ ie_option(ENABLE_ERROR_HIGHLIGHT "Highlight errors and warnings during compile t
...
+ie_dependent_option (ENABLE_PYTHON "enables ie python bridge build" ON "PYTHONLIBS_FOUND" OFF)
 
 find_package(PythonInterp 3 QUIET)
 ie_dependent_option (ENABLE_DOCS "Build docs using Doxygen" OFF "PYTHONINTERP_FOUND" OFF)
 
 # this option should not be a part of InferenceEngineDeveloperPackage
 # since wheels can be built only together with main OV build
-cmake_dependent_option (ENABLE_WHEEL "Build wheel packages for PyPi" OFF
-    "PYTHONINTERP_FOUND;CMAKE_SOURCE_DIR STREQUAL OpenVINO_SOURCE_DIR" OFF)
+cmake_dependent_option (ENABLE_WHEEL "Build wheel packages for PyPi" ON
```
Comment (ilya-lavrenov) [inline cmake/features.cmake:55]: should we `ENABLE_WHEEL` iff `ie_check_pip_package` has passed?

File: .github/workflows/code_style.yml [hunk 3]
```diff
@@ -92,7 +92,7 @@ jobs:
         run: |
           mkdir build
           cd build
-          cmake -DENABLE_INTEL_MYRIAD_COMMON=OFF ..
+          cmake -DENABLE_INTEL_MYRIAD_COMMON=OFF -DENABLE_PYTHON=OFF ..
```
[the remark this answers is not in the document]
Comment (slyubimt) [reply .github/workflows/code_style.yml]: disabled in [cmake/features.cmake](https://github.com/openvinotoolkit/openvino/pull/11278/files#diff-dd9d470bfd7d5c3c875bd6f1c182ac653ef5fb312ae28c24b8ea39c5074e7c8e)

File: cmake/features.cmake (hunk 2 above)
Comment (slyubimt) [reply cmake/features.cmake]: added ov_check_pip_packages function to check requirements.txt files
cmake/developer_package/IEDevScriptsConfig.cmake

File: src/bindings/python/wheel/CMakeLists.txt (hunk 1 above)
Comment (slyubimt) [reply src/bindings/python/wheel/CMakeLists.txt]: added ov_check_pip_packages function to check entire ${CMAKE_SOURCE_DIR}/src/bindings/python/wheel/requirements-dev.txt
[cmake/developer_package/IEDevScriptsConfig.cmake](https://github.com/openvinotoolkit/openvino/pull/11278/files#diff-bebb96ad9d754ddecb77341eb85eeb4d4736a623b1dfb8cdf6a9228545462fbc)

File: src/frontends/tensorflow/tests/CMakeLists.txt [hunk 4]
```diff
@@ -28,7 +30,7 @@ target_compile_definitions(${TARGET_NAME} PRIVATE -D TEST_TENSORFLOW_MODELS_DIRN
 # but models will not be generated and tests will fail
 # This is done this way for 'code style' and check cases - cmake shall pass, but CI machine doesn't need to have
 # 'tensorflow' installed to check code style
-if (tensorflow_FOUND)
+if (tensorflow)
     set(TEST_TENSORFLOW_MODELS ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${TEST_TENSORFLOW_MODELS_DIRNAME}/)
```
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply src/frontends/tensorflow/tests/CMakeLists.txt:34]: it's unified way to check python dependencies before using it.
It's already here in TF tests, so the PR is about unification for other places (specifically , checks dependencies before building OpenVINO wheel package)

[the remark this answers is not in the document]
Comment (rkazants) [reply src/frontends/tensorflow/tests/CMakeLists.txt:34]: Got it. Thanks!

[the remark this answers is not in the document]
Comment (slyubimt) [reply src/frontends/tensorflow/tests/CMakeLists.txt:34]: This is variable renaming due to ov_check_pip_package function API change. See more here
https://github.com/openvinotoolkit/openvino/pull/11278/files#diff-bebb96ad9d754ddecb77341eb85eeb4d4736a623b1dfb8cdf6a9228545462fbc

**Comment by ilya-lavrenov** [comment]
Replaced by https://github.com/openvinotoolkit/openvino/pull/12551

**Pull request closed without merging**